firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -151,22 +178,55 @@ public void finalizeReports(String currentSessionId) {
   }
 
   public List<CrashlyticsReport> loadFinalizedReports() {
-    final List<CrashlyticsReport> reports = new ArrayList<>();
-    final List<File> fatalReports = getAllFilesInDirectory(fatalReportsDirectory);
-    for (File reportFile : fatalReports) {
-      reports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
+    final List<CrashlyticsReport> allReports = new ArrayList<>();
+    final List<File> priorityReports = getAllFilesInDirectory(priorityReportsDirectory);
+    for (File reportFile : priorityReports) {
+      allReports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
     }
-    final List<File> nonFatalReports = getAllFilesInDirectory(nonFatalReportsDirectory);
-    for (File reportFile : nonFatalReports) {
-      reports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
+    final List<File> reports = getAllFilesInDirectory(reportsDirectory);
+    for (File reportFile : reports) {
+      allReports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
     }
-    return reports;
+    return allReports;
+  }
+
+  private static boolean isHighPriorityEvent(String eventFileName) {
+    return eventFileName.endsWith(PRIORITY_EVENT_SUFFIX);
+  }
+
+  private String generateEventFilename(boolean isHighPriority) {
+    final String eventNumber =
+        String.format(Locale.US, EVENT_COUNTER_FORMAT, eventCounter.getAndIncrement());
+    final String prioritySuffix = isHighPriority ? PRIORITY_EVENT_SUFFIX : NORMAL_EVENT_SUFFIX;
+    return EVENT_FILE_NAME_PREFIX + eventNumber + prioritySuffix;
   }
 
   private File getSessionDirectoryById(String sessionId) {
     return new File(openSessionsDirectory, sessionId);
   }
 
+  private static int trimEvents(File sessionDirectory, int maximum) {
+    final List<File> normalPriorityEventFiles =
+        getFilesInDirectory(
+            sessionDirectory, CrashlyticsReportPersistence::isNormalPriorityEventFile);
+    Collections.sort(normalPriorityEventFiles, CrashlyticsReportPersistence::oldestEventFileFirst);
+    return capFilesCount(normalPriorityEventFiles, maximum);
+  }
+
+  private static String eventNameOnly(String eventFileName) {
+    return eventFileName.substring(0, EVENT_NAME_LENGTH);
+  }
+
+  private static boolean isNormalPriorityEventFile(File dir, String name) {
+    return name.startsWith(EVENT_FILE_NAME_PREFIX) && !isHighPriorityEvent(name);
```

Comment:
Possibly a dumb optimization, but if !isHighPriorityEvent is more variable, and EVENT_FILE_NAME_PREFIX is expected 99% of the time, it might be worth putting the second check first to short circuit early more frequently.

Reply:
Why do we return an int if we never use it?

Reply:
:+1:

Reply:
Actually, after looking back at this - we need to filter out any non-event files *first*, so this needs to stay as it is.

Reply:
Rewrote these two methods though, and paired them, so they'd be easier to reason about.